src/backend/joanie/core/models/products.py

```diff
@@ -957,6 +974,14 @@ def submit_for_signature(self, user: User):
             contract = Contract(order=self, definition=contract_definition)
 
         if self.contract and self.contract.student_signed_on:
+            logger.error(
+                "Contract is already signed, cannot resubmit.",
```

Comment:
If my suggestion is accepted, the line 985 should be also updated.
```suggestion
                "Contract is already signed by the student, cannot resubmit.",
```